### DISCUSSION TITLE
Cannot Delete Device From Rack

### DISCUSSION THREAD

**Discussion by DrunkGorilla02** [netbox-community/netbox#14243] [Help Wanted!]
Hello everyone, 

I've been using netbox since 2022 and just encountered a Problem. I've wanted to delete a Device from a Server Rack, but got the error 

<class 'IndexError'>

list index out of range

Im usiung Python version: 3.10.12 and NetBox version: 3.5.7

any kind of Help is very appreciated since I've got a new bigger switch and wanted to delete the old one

**Comment by dreng** [comment]
Hi,

unfortunately you didn't provide the exact steps you followed in order to delete the device from the rack. This makes it somewhat difficult to help.

You can either:

- Edit the device and remove the rack position. This will make the device a non racked device but the device still belongs to the rack..
- Edit the device and remove the rack and the rack face. This means that the device is no longer part of the rack.
- Delete the entire device. This obviously means that the device no longer belongs to the rack.

**Reply by DrunkGorilla02** [reply]
Hello Dreng,

Thank you for the reply. I've tried to delete the whole device via the Devices Tab. Can I theoretically delete the Device from the Database directly?

**Reply by dreng** [reply]
In general, it is possible to delete the device from the database directly. The question is, does it help to solve your problem? NetBox itself does exactly this. Just give it a try.

The main issue here is the fact that it comes to an "Index out of range" error. There's a reason for that and I suspect there's some kind of inconsistent data in your database. Have you ever deleted some data directly from the database before?

**Reply by DrunkGorilla02** [reply]
Hello Dreng,

Thank you for responding.

I have not delete directly from the database **_yet_**.

**Comment by jeremystretch** [comment]
As it's unlikely you'll be able to easily identify steps to reproduce this condition, a stack trace would be helpful. You can enable these by setting `DEBUG = True` in `configuration.py` and restarting the NetBox service.

**Reply by DrunkGorilla02** [reply]
Hello Jeremy,

Thank you for Answering, I started the debug mode and got following Message:

Request Method: | POST
Jango Version: 4.1.10
Exception Type: IndexError
Exception value: list index out of range
Exception Location: /opt/netbox/venv/lib/python3.10/site-packages/django/db/models/query.py, line 446, in __getitem__
dcim.views.DeviceDeleteView
Raised During: /opt/netbox/venv/bin/python3
Python Version: 3.10.12
Python Path: ['/opt/netbox/netbox',
                      '/opt/netbox',  
                      '/opt/netbox/venv/bin', 
                      '/usr/lib/python310.zip',
                      '/usr/lib/python3.10', 
                      '/usr/lib/python3.10/lib-dynload', 
                      '/opt/netbox/venv/lib/python3.10/site-packages']
ServerTime: Thu, 16 Nov 2023 14:02:16 +0000
